### DISCUSSION TITLE
Add the ability to connect Fastify as a custom dev server

### DISCUSSION THREAD

**Discussion by sowtame** [web-infra-dev/rsbuild#2160] [Ideas]
here the main problems is how to connect dev middlewares and websocket in fastify for hmr

https://github.com/rspack-contrib/rspack-examples/pull/50

**Comment by 9aoy** [comment]
@sowtame I add an example for it. In fastify, you can get http server instance via `fastify.server`
 
https://github.com/rspack-contrib/rspack-examples/pull/54/files#diff-b75774b55d50e8e7e1b7443569feb1184f9d229a553e5f321e02dd8a8434ec53

**Reply by sowtame** [reply]
thank you, but is there a way that the server does not restart during hmr? (can reproduce in your example)
if you use express, there is no such thing
